lib/eyeshade/last_settlement_balance.rb

```diff
@@ -0,0 +1,40 @@
+module Eyeshade
+  class LastSettlementBalance < BaseBalance
+    attr_reader :date, :amount_settlement_currency, :timestamp, :settlement_currency
+
+    def initialize(rates, default_currency, transactions)
+      super(rates, default_currency)
+
+      last_settlement = calculate_last_settlement(transactions)
+      @amount_bat = last_settlement["amount_bat"]
+      @timestamp = last_settlement["timestamp"]
+      @settlement_currency = last_settlement["currency"]
+      @amount_settlement_currency = convert(@amount_bat, @settlement_currency)
+    end
+
+    private
+
+    def calculate_last_settlement(transactions)
+      return {} if transactions == []
+       # Find most recent settlement transaction
+      last_settlement_time = transactions.select { |transaction|
+        transaction["settlement_amount"].present?
+      }.last["created_at"].to_time # Eyeshade returns transactions ordered_by created_at
```

Comment:
this `.last["created_at"]` will be a problem when there are transactions, but none of them are settlement transactions

![image](https://user-images.githubusercontent.com/12549658/53376105-42497c80-392b-11e9-888d-5438f2429bab.png)

Reply:
I think we're fine. The prod data yields the following:
```
select count(*) from account_transactions
where created_at is null;
```
0

Reply:
replied via slack